### PR TITLE
🧹 Update AWS snapshot connection to use device connection.

### PR DESCRIPTION
**Pull request by preslavgerchev**
~~Putting this one in draft as it needs testing~~

### CODE REVIEW

File: providers/aws/provider/provider.go [hunk 1]
```diff
@@ -203,15 +203,6 @@ func (s *Service) connect(req *plugin.ConnectReq, callback plugin.ProviderCallba
 
 		case string(awsec2ebsconn.EBSConnectionType):
 			conn, err = awsec2ebsconn.NewAwsEbsConnection(connId, conf, asset)
-			if conn.Asset() != nil && len(conn.Asset().Connections) > 0 && conn.Asset().Connections[0].Options["mounted"] != "" {
```
Comment (preslavgerchev) [inline providers/aws/provider/provider.go:206]: im not sure why this was needed. the ebs conn seems to work without it just fine

Comment (vjeffrey) [reply providers/aws/provider/provider.go:206]: were you checking with builtin provider or not builtin? we saw some weird behaviour in the past that was diff between the two scenarios

Comment (preslavgerchev) [reply providers/aws/provider/provider.go:206]: built in. i can try testing with the provider binary separate too

Comment (preslavgerchev) [reply providers/aws/provider/provider.go:206]: Just tested, no issues with a non-built in provider

**Pull request merged**